projects/openapi-utilities/src/openapi3/json-path-interpreters.ts

```diff
@@ -0,0 +1,54 @@
+import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';
+import { IFact } from './sdk/types';
+
+// A trie datastructure keyed by json path parts
+export type FactTree = Record<
+  string,
+  { fact: IFact | null; children: FactTree }
+>;
+
+// We need a special json path decoder since the leading / is significant and points to the root
+const decodePathWithLeadingTrail = (pointer: string) => pointer.split('/').map(jsonPointerHelpers.unescape);
+
+export function constructFactTree(facts: IFact[]): FactTree {
+  const factTree: FactTree = {};
+  for (const fact of facts) {
+    let tree: FactTree = factTree;
+    const pathParts = decodePathWithLeadingTrail(fact.location.jsonPath);
+
+    for (let i = 0; i < pathParts.length; i++) {
+      const part = pathParts[i];
+      if (!tree[part]) {
+        tree[part] = { fact: null, children: {} };
+      }
+      // If it's the last iteration, it's the node where the fact should live
+      if (i === pathParts.length - 1) {
+        tree[part].fact = fact;
+      }
+      // Set up the next loop
+      tree = tree[part].children;
+    }
+  }
+
+  return factTree;
+}
+
+export function getFactForJsonPath(
```

Comment:
i need to think a little more about the implications if you pass a json path that is completely wrong in the current spec. The intention is that the json path is _in_ the spec/fact tree, but if it's not you could get a nonsense fact

Reply:
I bet we'll get a lot of things that are in the spec, but not in our fact tree. it makes sense -- we don't have facts for everything in the spec. 

Spectral is matching based on Json Path (the real one  https://goessner.net/articles/JsonPath/ ) so all the JSON Pointers we get out the other end should at least exist.